### PR TITLE
[ALLUXIO-1583] Add test coverage for S3InputStream.java

### PR DESCRIPTION
**Pull request by schernolyas**
https://alluxio.atlassian.net/browse/ALLUXIO-1583

### CODE REVIEW

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.s3;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertArrayEquals;
+
```
Comment (aaudiber) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: we usually put all static imports in one group: http://www.alluxio.org/docs/master/resources/eclipse_imports.png

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import static org.hamcrest.CoreMatchers.is;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
```
Comment (aaudiber) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: all third-part imports go in one group

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    mInputStreamSpy = Mockito.spy(new ByteArrayInputStream(new byte[] {1, 2, 3}));
+    mS3Service = Mockito.mock(S3Service.class);
+    mS3Object = Mockito.mock(S3Object.class);
+    when(mS3Object.getDataInputStream()).thenReturn(mInputStreamSpy);
+    when(mS3Service.getObject(BUCKET_NAME, OBJECT_KEY)).thenReturn(mS3Object);
+    mS3InputStream = new S3InputStream(BUCKET_NAME, OBJECT_KEY, mS3Service);
+  }
+
+  @After
+  public void tearDown() {}
```
Comment (aaudiber) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: if it doesn't do anything we can remove this

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+  /**
+   * Test of close method, of class S3InputStream.
+   */
+  @Test
+  public void close() throws IOException {
+    assertNotNull(mS3InputStream);
```
Comment (aaudiber) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: we set `mS3InputStream` in the `@Before` method, so we don't need to assert during the tests that it isn't null

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+  /**
+   * Test of read method, of class S3InputStream.
+   */
+  @Test
+  public void readWithArgs() throws Exception {
+    assertNotNull(mS3InputStream);
+    byte[] bytes = new byte[3];
+    int readedCount = mS3InputStream.read(bytes, 0, 3);
```
Comment (aaudiber) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: readedCount -> readCount

**Comment by schernolyas** [comment]
Hi @aaudiber !

I fixed your remarks. Recheck please

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  private S3InputStream mS3InputStream;
+  private S3Service mS3Service;
+  private S3Object mS3Object;
+  private InputStream mInputStreamSpy;
+
+  @Rule
+  public ExpectedException mExceptionRule = ExpectedException.none();
+
+  @BeforeClass
+  public static void setUpClass() throws ServiceException {}
```
Comment (aaudiber) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: we can remove this method

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+  @AfterClass
+  public static void tearDownClass() {}
```
Comment (aaudiber) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: we can remove this method

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java (hunk 6 above)
Comment (schernolyas) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: Hi @aaudiber !
fixed!

File: underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java (hunk 7 above)
Comment (schernolyas) [inline underfs/s3/src/test/java/alluxio/underfs/s3/S3InputStreamTest.java]: fixed!

**Comment by schernolyas** [comment]
Please recheck
